### ISSUE TITLE
fail2ban-2to3 currently fails against 0.10-full branch

### ISSUE THREAD

**Issue by ImNtReal** [sebres/fail2ban#10]
### Environment:
- Fail2Ban version (including any possible distribution suffixes): 0.10-full branch
- OS, including release name/version: arch linux, rolling
- [ ] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:
fail2ban-2to3 is failing with:
RefactoringTool: Can't parse fail2ban/server/actions.py: ParseError: bad input: type=1, value='if', context=(' ', (154, 44))

#### Steps to reproduce
git clone -b 0.10-full https://github.com/sebres/fail2ban.git
cd fail2ban
./fail2ban-2to3

#### Expected behavior
Success!

#### Observed behavior
Fail!

#### Any customizations done to /etc/fail2ban/ configuration
```
```

#### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

```
```

#### Relevant lines from monitored log files in question:

```
```

**Comment by sebres** [comment]
I cannot reproduce it. So I've get (tested with python 3.5 and 3.2):
```
...
RefactoringTool: Refactored fail2ban/server/action.py
RefactoringTool: Refactored fail2ban/server/actions.py
RefactoringTool: No changes to fail2ban/server/asyncserver.py
RefactoringTool: Refactored fail2ban/server/banmanager.py
....
Success!
```

Is the clone a clean copy (really unmodified)?
Did you see the same value?
```bash
$ openssl dgst -sha256 'fail2ban/server/actions.py'
SHA256(fail2ban/server/actions.py)= 6c5deb25aeb9f28c70d07e0a4a46dab7791c2f22b63aa65c50e5b9067a3e8ae4
```

$ python3 --version ?

Can you rename actions.py in actions.txt and provide it as attachment here?

**Comment by ImNtReal** [comment]
python --version
Python 3.6.0

SHA256(fail2ban/server/actions.py)= 6c5deb25aeb9f28c70d07e0a4a46dab7791c2f22b63aa65c50e5b9067a3e8ae4

**Comment by ImNtReal** [comment]
Interestingly, this seems to work fine:
`2to3 -w fail2ban/server/action.py
RefactoringTool: Skipping optional fixer: buffer
RefactoringTool: Skipping optional fixer: idioms
RefactoringTool: Skipping optional fixer: set_literal
RefactoringTool: Skipping optional fixer: ws_comma
RefactoringTool: Refactored fail2ban/server/action.py
--- fail2ban/server/action.py   (original)
+++ fail2ban/server/action.py   (refactored)
@@ -393,7 +393,7 @@
                while True:
                        repFlag = False
                        # substitute each value:
-                       for tag in tags.keys():
+                       for tag in list(tags.keys()):
                                # ignore escaped or already done (or in ignore list):
                                if tag in done: continue
                                value = orgval = str(tags[tag])
RefactoringTool: Files that were modified:
RefactoringTool: fail2ban/server/action.py`

**Comment by ImNtReal** [comment]
I just realized part of my problem was looking at action.py instead of actions.py while troubleshooting the issue. This is where the trouble is:

```
if name in self._actions:
                        self._actions[name].reload(**initOpts if initOpts else {})
```

**Comment by ImNtReal** [comment]
[actions.txt](https://github.com/sebres/fail2ban/files/775149/actions.txt)

**Comment by sebres** [comment]
> actions.txt

You've the same SHA value :) I'll delete the attachment...

I think this may be a bug of py-2to3 of python3.6 (or rather it is no more backwards compatible).

ATM no python3.6 handy to test it.
Can you try to change and check:
```diff
- self._actions[name].reload(**initOpts if initOpts else {})
+ self._actions[name].reload(**(initOpts if initOpts else {}))
```

**Comment by ImNtReal** [comment]
I figured the attachment was pointless, then, but wanted to make sure. Replacing the line appears to correct the problem with 2to3. I'm going to try with the full -2to3 script to make double sure.

**Comment by ImNtReal** [comment]
Yep. That one line was the only one that needed to be changed for the script to complete.

**Comment by ImNtReal** [comment]
Building from the latest commit on Python 3.6 works great.

**Comment by sebres** [comment]
Tested against newest python3.6, found another incompatibilities...
Fixed now in py-3.6-compat branch and merged in 0.10-full (see fa20b0aff2be0f0fff5ed796cd2bd31ec79f275f)
